### PR TITLE
Add RMF deep link to Accessibility/voice search

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206878119354754/f

### Description
Adds a deep link to the accessibility screen (If voice search is supported), highlighting the voice search toggle.

### Steps to test this PR

_On a device that supports voice search, point to the RMF config linked in the task_
- [ ] Complete onboarding.
- [ ] Verify that the RMF dialog is show and is correct.
- [ ] Click “Add Private Voice Search”.
- [ ] Verify that the Accessibility screen is shown.
- [ ] Verify that the voice search toggle is highlighted. 

### UI changes

https://github.com/duckduckgo/Android/assets/3471025/e825c651-fd29-4977-adaf-fcc13c9c6ac0

### CODE REVIEW

**Comment by joshliebe** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4394?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4394** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4394?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4393** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4393?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joshliebe and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt [hunk 1]
```diff
@@ -67,6 +72,37 @@ class AccessibilityActivity : DuckDuckGoActivity() {
         setContentView(binding.root)
         setupToolbar(toolbar)
         observeViewModel()
+        scrollToHighlightedItem()
+    }
+
+    private fun scrollToHighlightedItem() {
+        intent.getActivityParams(AccessibilityScreenParams::class.java)?.let { params ->
+            if (params.highlightedItem == "voiceSearch") {
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt:80]: NIT: to put "voiceSearch" into a const

File: app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt [hunk 2]
```diff
@@ -67,6 +72,37 @@ class AccessibilityActivity : DuckDuckGoActivity() {
...
+
+    private fun scrollToVoiceSearchToggle() {
+        val scrollTo = binding.voiceSearchToggle.top
+        binding.scrollView.smoothScrollTo(0, scrollTo)
+    }
+    private fun highlightVoiceSearchToggle() {
+        val highlightColor = getColorFromAttr(com.duckduckgo.mobile.android.R.attr.daxColorContainer)
+        val transparentColor = ContextCompat.getColor(applicationContext, android.R.color.transparent)
+
+        val flickerDuration = 300L
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt:97]: NIT: to move those values into const

File: app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt [hunk 3]
```diff
@@ -67,6 +72,37 @@ class AccessibilityActivity : DuckDuckGoActivity() {
...
+        val totalAnimationDuration = flickerDuration * 5
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt:98]: move 5 to a const that explains what does 5 mean.

File: app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt [hunk 4]
```diff
@@ -38,6 +42,7 @@ import timber.log.Timber
 
 @InjectWith(ActivityScope::class)
 @ContributeToActivityStarter(AccessibilityScreenNoParams::class)
+@ContributeToActivityStarter(AccessibilityScreenParams::class, screenName = "accessibility")
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/accessibility/AccessibilityActivity.kt:45]: we can add "accessibility" also to the other `@ContributeToActivityStarter`, so we can have the navigation with empty params also exposed.

**Review by cmonfortep — APPROVED** [review]
LGTM just minor comments. thanks!

**Pull request merged**